### DISCUSSION TITLE
Problem While compiling PrimeCPP/solution 2

### DISCUSSION THREAD

**Discussion by pinguimfu** [PlummersSoftwareLLC/Primes#975] [Q&A]
steps taken:

clone git repo inside a ubuntu 22.04 machine running amd64 hardware (Ryzen 3 3200g)
cd into said directory
apt install dependencies (?) (sudo apt install clang build-essentials libc++-dev  libstdc++-dev)
run run.sh gives me this output:

```
gabriel@blackbox:~/Programas/Primes/PrimeCPP/solution_2$ ./run.sh 
PrimeCPP_PAR.cpp:7:10: fatal error: 'chrono' file not found
#include <chrono>
         ^~~~~~~~
1 error generated.
./run.sh: linha 6: ./primes_par.exe: Arquivo ou diretório inexistente
gabriel@blackbox:~/Programas/Primes/PrimeCPP/solution_2$ 
```

what iam doing wrong?

Thanks for the help

**Comment by rbergen** [comment]
I'm converting this to a Q&A discussion. This is not a project issue but a local problem; CI builds fine, and it works on my machine - and that of at least one other person.

**Comment by rbergen** [comment]
Looking at your "sudo apt install" I can tell you that you can deselect build-essentials, libc++-dev and libstdc++-dev, at least for this solution. Manually installing clang is enough; it will pull in everything else it needs. You can try running the following commands to remove the manual install flags on the other packages and clean up whatever is actually not needed:

```shell
sudo apt-mark auto build-essentials libc++-dev libstdc++-dev
sudo apt autoremove
```

There's no reason to assume this will solve your problem, but it'll remove bloat from your setup.

**Comment by pinguimfu** [comment]
i tried on my work machine with the same setup and here it worked first time... it must be something with my machine at home... sorry for the bother...

just to note - the packages i installed for it to sucessfully compile are as follows:

sudo apt install clang libc++-dev libstdc++-12-dev

**Reply by rbergen** [reply]
No problem. I hope you manage to also get your home machine working!
